Middleware/SymfonyRouteResolver.php

```diff
@@ -95,7 +100,7 @@ public function __construct(
      */
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
-        $symfonyRequest = $this->httpFoundationFactory->createRequest($request);
+        $symfonyRequest = $this->httpFoundationFactory->createRequest($this->prepareForSymfonyRequest($request));
```

Comment:
I dont understand the reason why calling it here. In this case we are catching a preparing a symfony route, which doesn't have the context of a page?

Reply:
You are right! I made changes, so this is only done, when it makes sense. Additionally i changed the way the locale is determined for the request.